src/pytopmod/core/dcel/operators.py

```diff
@@ -0,0 +1,277 @@
+        else edge_1_node.face_2_key
+    )
+
+    # Set the vertex and edge information for the new edge.
+    # 2 - Create a new edge node.
+    new_edge_key = mesh.create_edge(
+        vertex_1_key, vertex_2_key, face_1_key, face_2_key, edge_1_2_key, edge_2_2_key
+    )
+    new_edge_node = mesh.edge_nodes[new_edge_key]
+
+    # Non-cofacial insertion.
+    if face_1_key != face_2_key:
+        # 3.1 - Update the face information.
+        # Create a new face.
+        new_face_key = mesh.create_face()
+        # Traverse the corners' faces and replace face_1 and face_2 by the new face.
+        for edge_key in list(
+            itertools.chain(face_trace(mesh, face_1_key), face_trace(mesh, face_2_key))
+        ):
+            edge_node = mesh.edge_nodes[edge_key]
+            if edge_node.face_1_key in (face_1_key, face_2_key):
+                edge_node.face_1_key = new_face_key
+            if edge_node.face_2_key in (face_1_key, face_2_key):
+                edge_node.face_2_key = new_face_key
+        # Set the face information for the new edge.
+        new_edge_node.face_1_key = new_face_key
+        new_edge_node.face_2_key = new_face_key
+        # Delete the old faces.
+        mesh.delete_face(face_1_key)
+        mesh.delete_face(face_2_key)
+
+        # 3.2 - Update the edge information.
+        if edge_1_node.vertex_1_key == vertex_1_key:
+            edge_1_node.vertex_1_next_key = new_edge_key
+        else:
+            edge_1_node.vertex_2_next_key = new_edge_key
+        if edge_2_node.vertex_1_key == vertex_2_key:
+            edge_2_node.vertex_1_next_key = new_edge_key
+        else:
+            edge_2_node.vertex_2_next_key = new_edge_key
+
+    # Cofacial insertion.
+    else:
+        # 4.1 - Update the edge information.
+        if edge_1_node.vertex_1_key == vertex_1_key:
+            edge_1_node.vertex_1_next_key = new_edge_key
+        else:
+            edge_1_node.vertex_2_next_key = new_edge_key
+        if edge_2_node.vertex_1_key == vertex_2_key:
+            edge_2_node.vertex_1_next_key = new_edge_key
+        else:
+            edge_2_node.vertex_2_next_key = new_edge_key
+
+        # 4.2 - Update the face information.
+        # Create two new faces.
+        new_face_1_key = mesh.create_face()
+        new_face_2_key = mesh.create_face()
+
+        # Starting from one direction of the new edge, replace face_1 by new_face_1.
+        for edge_key in list(face_trace(mesh, face_1_key, start_edge_key=edge_1_2_key)):
+            edge_node = mesh.edge_nodes[edge_key]
+            if edge_node.face_1_key == face_1_key:
+                edge_node.face_1_key = new_face_1_key
+            else:
+                edge_node.face_2_key = new_face_1_key
+
+        # Starting from the opposite direction, replace face_1 by new_face_2.
+        for edge_key in list(face_trace(mesh, face_1_key, start_edge_key=edge_2_2_key)):
+            edge_node = mesh.edge_nodes[edge_key]
```

Comment:
suggestion: This is similar to the previous step, we can put it in an helper function and call something like

```python
# Starting from one direction of the new edge, replace face_1 by new_face_1.
 for edge_key in list(face_trace(mesh, face_1_key, start_edge_key=edge_1_2_key)):
      updateFaces(mesh, edge_key, face_1_key, new_face_1_key)

# Starting from the opposite direction, replace face_1 by new_face_2.
for edge_key in list(face_trace(mesh, face_1_key, start_edge_key=edge_2_2_key)):
      updateFaces(mesh, edge_key, face_1_key, new_face_2_key)
```

Reply:
See previous comment